Gems/Atom/RPI/Code/Source/RPI.Public/Pass/AttachmentReadback.cpp

```diff
@@ -454,6 +431,29 @@ namespace AZ
 
         void AttachmentReadback::CopyExecute(const RHI::FrameGraphExecuteContext& context)
         {
+            uint32_t readbackBufferCurrentIndex = m_readbackBufferCurrentIndex;
```

Comment:
Any specific reason for moving this here?

Reply:
Yes, in order to wait for the fence on the correct device, we need to know the device index and we only get that once the frame graph is being executed (we get the device index from the context here). We only want to wait for the fence on the device that this scope actually runs on, since no fences will be triggered for the other devices.

The reason we put this here at all is that if the scheduler gets the capability of choosing which device a scope runs on, we do not know the device index during the preparation yet, since the decision is made during the compilation stage.